domain/ide/workspace/folder.go

```diff
@@ -479,46 +479,42 @@ func appendTestResults(sic snyk.SeverityIssueCounts, results []json_schemas.Test
 	return results
 }
 
-func (f *Folder) FilterAndPublishDiagnostics(p *product.Product) {
-	productIssuesByFile, err := f.getDelta(f.IssuesByProduct(), p)
+func (f *Folder) FilterAndPublishDiagnostics(p product.Product) {
+	issueByProduct := f.IssuesByProduct()
+	if len(issueByProduct[p]) == 0 {
+		return
+	}
+
+	productIssuesByFile, err := f.getDelta(issueByProduct, p)
 	if err != nil {
 		// Error can only be returned from delta analysis. Other non delta scans are skipped with no errors.
 		deltaErr := fmt.Errorf("couldn't determine the difference between current and base branch for %s scan. %w", p.ToProductNamesString(), err)
-		f.sendScanError(*p, deltaErr)
+		f.sendScanError(p, deltaErr)
 		return
 	}
-	if p != nil {
-		filteredIssues := f.filterDiagnostics(productIssuesByFile[*p])
-		f.publishDiagnostics(*p, filteredIssues)
-	} else {
-		for p, pIssues := range productIssuesByFile {
-			filteredIssues := f.filterDiagnostics(pIssues)
-			f.publishDiagnostics(p, filteredIssues)
-		}
-	}
+	filteredIssues := f.filterDiagnostics(productIssuesByFile[p])
```

Comment:
we need to send publishDiagnostics with an empty diagnostics array for files that don't have diagnostics anymore. so if after filtering, no issues are left for a file that has issues, we need to send publish diagnostics.